client/src/details/UrlBuilder.cpp

```diff
@@ -23,6 +23,33 @@ using namespace SFS::details;
 
 namespace
 {
+
+struct CurlEscapedUrlString

```

Comment:
this should probably publically derive from a std::unique_ptr with a custom deleter that calls curl_free so that way we have a reusable object for dealing with curl_free'd objects, including this one.